include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -82,10 +82,15 @@ class pairwise_alignment_algorithm_banded :
         lower_diagonal = band.lower_diagonal.get();
         upper_diagonal = band.upper_diagonal.get();
 
-        if (upper_diagonal < lower_diagonal || upper_diagonal < 0 || lower_diagonal > 0)
+        // Band is invalid if ...
+        bool const invalid_band = upper_diagonal < lower_diagonal || // upper diagonal is smaller than lower diagonal,
+                                  (upper_diagonal < 0 && !this->first_column_is_free) || // band starts in first column but does not use free ends,
+                                  (lower_diagonal > 0 && !this->first_row_is_free); // band starts in first row but does not use free ends.
+
+        if (invalid_band)
             throw invalid_alignment_configuration{"The selected band [" + std::to_string(lower_diagonal) + ":" +
-                                                  std::to_string(upper_diagonal) + "] is not supported because it "
-                                                  "does not completely encloses both sequences."};
+                                                  std::to_string(upper_diagonal) + "] is not valid for the "
+                                                  "configured alignment because the optimum cannot be computed."};
```

Comment:
This conflicts with https://github.com/seqan/seqan3/pull/1935/files#diff-a374649bb6a2078f62e2bbc24c7014ffL77-L89, right?

What is your ultimate version?

Reply:
The ultimate version is to not throw at all inside of the alignment algorithm but rather set an error state inside of the alignment result. Then the user can ask if an error state was set or in case he calls a member function while the error state is set it will throw. This allows us to not worry about correctly populating exceptions in threaded code, which is everything but trivial. Then the user has to make sure his code is handling the exceptions correctly. Until then, I want to put this into common structures to not repeat the code.

Reply:
I agree, but I will make it more verbose in #1935 because it has a more appropriate structure to do this.